manifests/state-doca-telemetry-service/0010-doca-telemetry-service.yaml

```diff
@@ -0,0 +1,140 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: doca-telemetry-service
+  namespace: {{ .RuntimeSpec.Namespace }}
+spec:
+  # for DaemonSet
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: doca-telemetry
+  updateStrategy:
+    type: RollingUpdate
+    # maxUnavailable: 25%
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: doca-telemetry
+    spec:
+      # hostNetwork is required to expose sysfs counters folder
+      hostNetwork: true
+      {{- if .NodeAffinity }}
+      affinity:
+        nodeAffinity:
+          {{- .NodeAffinity | yaml | nindent 10 }}
+      {{- end }}
+      {{- if .CrSpec.ImagePullSecrets }}
+      imagePullSecrets:
+      {{- range .CrSpec.ImagePullSecrets }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      containers:
+      - name: doca-telemetry-service
+        image: {{ .CrSpec.Repository }}/{{ .CrSpec.Image }}:{{ .CrSpec.Version }}
+        {{- with .RuntimeSpec.ContainerResources }}
+        {{- with index . "nic-feature-discovery" }}
+        resources:
+          {{- if .Requests }}
+          requests:
+            {{ .Requests | yaml | nindent 14}}
+          {{- end }}
+          {{- if .Limits }}
+          limits:
+            { { .Limits | yaml | nindent 14 }}
+          {{- end }}
+        {{- end }}
+        {{- end }}
+        volumeMounts:
+        - name: doca-telemetry-service-configmap
+          mountPath: /configmap
+        - name: pod-device-resources
+          mountPath: /var/lib/kubelet/pod-resources
+        - name: cni
+          mountPath: /var/run/k8s.cni.cncf.io/devinfo/cni
+        env:
+        - name: MY_POD_UID
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.uid
+        - name: PROMETHEUS_XCSET_JOIN_FIELDS
+          value: "hca"
+        - name: PROMETHEUS_XCSET_MANDATORY_TYPES
+          value: "counters,pod_resources_event"
+        command: [ "/bin/bash", "-c", "rm -rf /config/* && DTS_CONFIG_DIR=host /usr/bin/telemetry-init.sh && /usr/bin/telemetry-run.sh" ]
+        securityContext:
+          privileged: true
+      volumes:
+        - name: doca-telemetry-service-configmap
+          configMap:
+            name: {{ .ConfigMapName }}
+        - name: pod-device-resources
+          hostPath:
+            path: /var/lib/kubelet/pod-resources
+        - name: cni
+          hostPath:
+            path: /var/run/k8s.cni.cncf.io/devinfo/cni/
```

Comment:
This may not exist which i guess is ok.

if it becomes available later will this behave as expected ?

Reply:
I'm really not sure - the DTS container will fail silently in most cases where things don't work at startup - this might be one of those cases.

Reply:
I'll follow up and try to figure out if this prevents the pod devices collector from running.

Reply:
I'm going to push this beyond the scope of this PR - will tackle another round of review of the yaml and use cases in a follow-up.